homeassistant/components/media_player/spotify.py

```diff
@@ -328,3 +368,44 @@ def supported_features(self):
     def media_content_type(self):
         """Return the media type."""
         return MEDIA_TYPE_MUSIC
+
+
+class SpotifyPlaylistView(HomeAssistantView):
+    """View to retrieve Spotify Playlists."""
+
+    url = '/api/spotify/playlists'
+    name = "api:spotify/playlists"
+
+    @callback
+    def get(self, request):
+        """Retrieve playlists."""
+        return self.json(request.app['hass'].data[DOMAIN].playlists)
+
+
+class PlaylistData:
+    """Class to hold Spotify playlist data."""
+
+    def __init__(self, hass):
+        """Initialize the Spotify playlist store list."""
+        self.hass = hass
+        self.playlists = {}
+
+    def update(self, playlist=None):
+        """Add a playlist."""
+        if playlist is not None:
+            self.playlists[playlist['id']] = playlist
+        self.save()
+        self.playlists = self.load()
+
+    def load(self):
+        """Load the playlists."""
+        path = self.hass.config.path(PERSISTENCE)
+        if not os.path.isfile(path):
+            return {}
+        with open(path) as file:
+            return json.loads(file.read())
+
+    def save(self):
+        """Save playlists."""
+        with open(self.hass.config.path(PERSISTENCE), 'wt') as file:
```

Comment:
Over complicated and "file" is a dangerous variable name, I suggest:

```python
with open(self.hass.config.path(PERSISTENCE), 'wt') as p_file:
    json.dump(self.playlist,p_file)
```

Reply:
Can you suggest how you would simplify it?

Reply:
look at the code in my comment ...

Reply:
need to indent lines 20:21 and 25:26, but that looks good